rusk/src/lib/chain/rusk.rs

```diff
@@ -267,64 +248,51 @@ impl Rusk {
             }
         }
 
-        let commit_id = session.commit()?;
-        inner.current_commit = commit_id;
-
-        // Delete all commits except the previous base commit, and the current
-        // commit
-        let mut delete_commits = inner.vm.commits();
-        delete_commits.retain(|c| {
-            c != &inner.current_commit
-                && c != &inner.base_commit
-                && c != &current_commit
-        });
-        for commit in delete_commits {
-            inner.vm.delete_commit(commit)?;
-        }
+        let commit = session.commit()?;
+        self.set_base_and_delete(commit);
 
         let commit_id_path = to_rusk_state_id_path(&self.dir);
-        fs::write(commit_id_path, commit_id)?;
-
-        inner.base_commit = commit_id;
+        fs::write(commit_id_path, commit)?;
 
         Ok((spent_txs, verification_output))
     }
 
     pub fn revert(&self, state_hash: [u8; 32]) -> Result<[u8; 32]> {
-        let mut inner = self.inner.lock();
+        let mut tip = self.tip.write();
 
-        let commits = &inner.vm.commits();
+        let commits = self.vm.commits();
         if !commits.contains(&state_hash) {
             return Err(Error::CommitNotFound(state_hash));
         }
 
-        inner.current_commit = state_hash;
-        Ok(inner.current_commit)
+        tip.current = state_hash;
+        Ok(tip.current)
     }
 
     pub fn revert_to_base_root(&self) -> Result<[u8; 32]> {
         self.revert(self.base_root())
     }
 
     /// Perform an action with the underlying data structure.
-    pub fn with_inner<'a, F, T>(&'a self, closure: F) -> T
+    ///
+    /// This should **not be used** internally, to avoid locking the structure
+    /// for too long of a period of time.
+    pub fn with_tip<'a, F, T>(&'a self, closure: F) -> T
```

Comment:
Can you put this behind `#[cfg(feature = "testwallet")]`?

So to have it available only to tests